consensus/src/lib.rs

```diff
@@ -76,73 +75,120 @@ where
         + 'static,
     Ctx::Future: Send + 'static,
 {
-    let tx_svc = transactions::TxVerifierService::new(database.clone());
-    let block_svc = block::BlockVerifierService::new(ctx_svc, tx_svc.clone(), database);
+    let tx_svc = TxVerifierService::new(database.clone());
+    let block_svc = BlockVerifierService::new(ctx_svc, tx_svc.clone(), database);
     Ok((block_svc, tx_svc))
 }
 
+/// An internal trait used to represent a database so we don't have to write [`tower::Service`] bounds
+/// everywhere.
 pub trait Database:
-    tower::Service<DatabaseRequest, Response = DatabaseResponse, Error = tower::BoxError>
+    tower::Service<
+    DatabaseRequest,
+    Response = DatabaseResponse,
+    Error = tower::BoxError,
+    Future = Self::Future2,
+>
 {
+    type Future2: Future<Output = Result<Self::Response, Self::Error>> + Send + 'static;
 }
 
 impl<T: tower::Service<DatabaseRequest, Response = DatabaseResponse, Error = tower::BoxError>>
     Database for T
+where
+    T::Future: Future<Output = Result<Self::Response, Self::Error>> + Send + 'static,
 {
+    type Future2 = T::Future;
 }
 
+/// An extended block header.
 #[derive(Debug, Copy, Clone)]
 pub struct ExtendedBlockHeader {
+    /// The blocks major version.
     pub version: HardFork,
+    /// The blocks vote.
     pub vote: HardFork,
 
+    /// The blocks timestamp.
     pub timestamp: u64,
+    /// The blocks cumulative difficulty.
     pub cumulative_difficulty: u128,
 
+    /// The blocks weight.
     pub block_weight: usize,
+    /// The blocks long term.
     pub long_term_weight: usize,
 }
 
+/// A database request to the database [`tower::Service`]
 #[derive(Debug, Clone)]
 pub enum DatabaseRequest {
+    /// A block extended header request.
+    /// Must return: [`DatabaseResponse::BlockExtendedHeader`]
```

Comment:
There's already docs on `cuprate_types::service::{Response,Request}`, maybe copy those?

Btw, cargo docs will render this as a single line, there needs to be a space between lines.

Actually, what is the plan here? Why not use/edit the `Request/Response` from `types`?

Reply:
In another PR these will be replaced with the enums in `cuprate_types` I didn't want to do it here as the DB would need updating